broadinstitute/single_cell_portal_core
# Fix initial width of violin plot (SCP-4945)

This lets users see the distributions more clearly in violin plots, by fixing a bug in their initial display.

Previously, the violin plot was quite narrow upon its first rendering.  The plot could resize to expected proportions in response to changes in its containers dimensions, but its initial display (the most important) was flawed.  This was due to how some layout properties interacted with the plot is written to a hidden display at first.

Now, violin plots always have a reasonable width.  This applies to both single-gene view, which is the most case common, as well as the "Collapse by" consensus display for >= 3 genes, which is much less common.

### Bug

https://user-images.githubusercontent.com/1334561/230455921-849c5069-26c5-4743-8ef9-1700fa7b40b7.mov

### Fix

https://user-images.githubusercontent.com/1334561/230456063-b8ab11dc-45a5-4d9c-acab-81183d5f4d8d.mov

### Test

* Go to initialized study
* Search 1 gene
* Click "Distribution" tab
* Confirm violin plot width fills available space
* Search 3 genes (e.g. PTEN, PIK3CA, AKT1)
* In "Collapse genes by" menu, select a metric, e.g. "Mean"
* Confirm violin plot width fills available space, and caption displays at bottom (e.g. "Mean expression of PTEN, PIK3CA, AKT1")


This satisfies SCP-4945.
